### PR TITLE
feat: add futurex api client

### PR DESCRIPTION
**Pull request by andrey-canon**
## Description

This includes the api client to communicate with futurex services, this just include one case, that is the enrollment progress

## Testing instructions

it is not possible since we don't have credentials

## Additional information

Epic link https://edunext.atlassian.net/jira/software/c/projects/NELP/issues/NELP-371

## Checklist for Merge

- [ ] Tested in a remote environment
- [ ] Updated documentation
- [ ] Rebased master/main
- [ ] Squashed commits

### CODE REVIEW

File: requirements/pip-tools.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #
-# This file is autogenerated by pip-compile with Python 3.8
+# This file is autogenerated by pip-compile with Python 3.10
```
Comment (johanv26) [inline requirements/pip-tools.txt:2]: Do you know why this is using python 3.10?

File: eox_nelp/course_api/serializers.py [hunk 2]
```diff
@@ -22,6 +23,7 @@ class NelpCourseDetailSerializer(CourseDetailSerializer):  # pylint: disable=abs
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
+        FuturexApiClient()
```
Comment (johanv26) [inline eox_nelp/course_api/serializers.py:26]: Why this is added to nelp course detail serializer?

File: eox_nelp/api_clients/futurex.py [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+            data<Dict>: request body as dictionary.
+
+        Return:
+            Dictionary: Empty dictionary or json response.
+        """
+        url = f"{self.base_url}/{path}"
+
+        response = self.session.post(url=url, json=data)
+
+        if response.status_code == status.HTTP_200_OK:
```
Comment (johanv26) [inline eox_nelp/api_clients/futurex.py:84]: 201?

Comment (andrey-canon) [reply eox_nelp/api_clients/futurex.py:84]: https://drive.google.com/file/d/16PXRFCFZuRQMjleQqkm_qSKSfddpwT8g/view

File: requirements/pip-tools.txt (hunk 1 above)
Comment (andrey-canon) [reply requirements/pip-tools.txt:2]: I run this in my local environment

File: eox_nelp/course_api/serializers.py (hunk 2 above)
Comment (andrey-canon) [reply eox_nelp/course_api/serializers.py:26]: my mistake

File: eox_nelp/settings/common.py [hunk 4]
```diff
@@ -28,5 +28,9 @@ def plugin_settings(settings):
     settings.EOX_NELP_MFE_CONFIG_VIEW = 'eox_nelp.edxapp_wrapper.backends.mfe_config_view_m_v1'
     settings.EOX_NELP_COURSE_API = 'eox_nelp.edxapp_wrapper.backends.course_api_m_v1'
 
+    settings.FUTUREX_API_URL = 'https://testing-site.com'
```
[the remark this answers is not in the document]
Comment (andrey-canon) [reply eox_nelp/settings/common.py:31]: yes, they would, all these information is sensitive therefore we can not include values similar to the real values

**Review by johanv26 — APPROVED** [review]
I think that everything is ok, may be in the future we would check this https://github.com/eduNEXT/eox-nelp/pull/35#discussion_r1144994993.
But at the moment, is a good client for me.

File: eox_nelp/api_clients/futurex.py [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+        Args:
+            enrollment_data<Dict>: Information about a specific progress.
+
+        Returns:
+            response<Dict>: requests response as dictionary.
+
+        Raise:
+            Basic exception when the enrollment data is not completed.
+        """
+        path = "enrollment-progress"
```
Comment (johanv26) [inline eox_nelp/api_clients/futurex.py:136]: Is very strange for me to add a path without left `/` but apart from this is working.

**Pull request merged**